src/component/admin/users/UsersList/UsersList.tsx

```diff
@@ -116,136 +97,233 @@ const UsersList = ({ search }: IUsersListProps) => {
+                width: '40%',
+                Cell: HighlightCell,
+            },
+            {
+                id: 'username',
+                Header: 'Username',
+                accessor: (row: any) => row.username || row.email,
+                width: '40%',
+                Cell: HighlightCell,
+            },
+            {
+                id: 'role',
+                Header: 'Role',
+                accessor: (row: any) =>
+                    roles.find((role: IRole) => role.id === row.rootRole)
+                        ?.name || '',
+                disableGlobalFilter: true,
+            },
+            {
+                id: 'last-login',
+                Header: 'Last login',
+                accessor: (row: any) => row.seenAt || '',
+                Cell: ({ row: { original: user } }: any) => (
+                    <TimeAgoCell value={user.seenAt} emptyText="Never logged" />
+                ),
+                sortType: 'date',
+                disableGlobalFilter: true,
+            },
+            {
+                Header: 'Actions',
+                id: 'Actions',
+                align: 'center',
+                Cell: ({ row: { original: user } }: any) => (
+                    <Box sx={{ display: 'flex', justifyContent: 'center' }}>
+                        <PermissionIconButton
+                            data-loading
+                            onClick={() => {
+                                navigate(`/admin/users/${user.id}/edit`);
+                            }}
+                            permission={ADMIN}
+                            tooltipProps={{
+                                title: 'Edit user',
+                            }}
+                        >
+                            <Edit />
+                        </PermissionIconButton>
+                        <PermissionIconButton
+                            data-loading
+                            onClick={openPwDialog(user)}
+                            permission={ADMIN}
+                            tooltipProps={{
+                                title: 'Change password',
+                            }}
+                        >
+                            <Lock />
+                        </PermissionIconButton>
+                        <PermissionIconButton
+                            data-loading
+                            onClick={openDelDialog(user)}
+                            permission={ADMIN}
+                            tooltipProps={{
+                                title: 'Remove user',
+                            }}
+                        >
+                            <Delete />
+                        </PermissionIconButton>
+                    </Box>
+                ),
+                width: 100,
+                disableSortBy: true,
+            },
+        ],
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [loading, navigate, isBillingUsers]
```

Comment:
Was having some trouble with these dependencies. For example, `roles` was suggested as a dependency but when adding it I got an infinite (render?) loop React error. 
These current dependencies make logical sense to me, but maybe someone more experienced with React has a better solution here.

Reply:
Yeah, you were right to add the dependency. Would be good to get rid of all these annotations.

The problem seems to be that the roles object is unstable. `roles` is a new empty array on each render (until the fetch completes). When `roles` change, `columns` change, and then `useTable` probably triggers another render. 

Could change `useUsers` to something like this:

```ts
import useSWR from 'swr';
import { useMemo } from 'react';
import { formatApiPath } from 'utils/formatPath';
import handleErrorResponses from '../httpErrorResponseHandler';

export const useUsers = () => {
    const { data, error, mutate } = useSWR(
        formatApiPath(`api/admin/user-admin`),
        fetcher
    );

    return useMemo(
        () => ({
            users: data?.users || [],
            roles: data?.rootRoles || [],
            loading: !error && !data,
            refetch: () => mutate,
            error,
        }),
        [data, error, mutate]
    );
};

const fetcher = (path: string) => {
    return fetch(path)
        .then(handleErrorResponses('Users'))
        .then(res => res.json());
};
```

Reply:
Wow, seems to have been exactly that - This works perfectly, thanks! 😄